src/index.ts

```diff
@@ -421,9 +427,39 @@ export class Upload extends Pumpify {
     if (this.origin) {
       reqOpts.headers!.Origin = this.origin;
     }
+    const uri = await retry(
+      async (bail: (err: Error) => void) => {
+        try {
+          const res = await this.makeRequest(reqOpts);
+          return res.headers.location;
+        }
+        catch (e){
+          const apiError = {
+            code: e.response.status,
+            name: e.response.statusText,
+            message: e.response.statusText,
+            errors: [
+              {
+                reason: e.response.statusText
+              }
+            ]
+          }
+          if (this.retryLimit > 0 && this.retryableErrorFn!(apiError)) {
+            throw new Error();
+          }
+          else {
+            return bail(new Error(`Failed with error ${e.response}`));
```

Comment:
If possible, this should probably return an ApiError up to bail so that we can include the code as well as the message.

Reply:
i'm going to return `e` raw to preserve previous behavior